### PR TITLE
python3Packages.mocket: Resolve test failures

### PR DESCRIPTION
**Pull request by hesiod**
###### Description of changes

Add pytest-asyncio in order to allow async tests to run. They were previously silently skipped with a warning.

Disable three more tests (test_http*_session) that depend on Internet connectivity and thus fail in sandboxed builds.

This should some Hydra failures, e.g. this one: https://hydra.nixos.org/build/208832590

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/mocket/default.nix [hunk 1]
```diff
@@ -72,6 +74,9 @@ buildPythonPackage rec {
     "test_real_request_session"
     "test_asyncio_record_replay"
     "test_gethostbyname"
+    "test_http_session"
+    "test_https_session"
+    "test_httprettish_session"
```
Comment (mweinelt) [inline pkgs/development/python-modules/mocket/default.nix:79]: Set `__darwinAllowLocalNetworking = true;` instead.

Comment (hesiod) [reply pkgs/development/python-modules/mocket/default.nix:79]: Thanks, I didn't know about that! However, I'm not sure if it's even a Darwin-specific issue anymore -see below

**Comment by hesiod** [comment]
The tests also fail on x86_64-linux for me (even with the sandbox disabled).
At this point I think it might actually be due to changes to `asyncio.get_event_loop()`, evident in this warning:
```
tests/main/test_http_aiohttp.py::AioHttpEntryTestCase::test_http_session
  /build/mocket-3.10.9/tests/main/test_http_aiohttp.py:35: DeprecationWarning: There is no current event loop
    loop = asyncio.get_event_loop()
```
(printed a couple of times)

That would mean it's simply an upstream Python 3.11 compatibility issue.
Running nixpkgs-review reveals similar failures in reverse dependencies of mocket.

**Comment by hesiod** [comment]
Result of `nixpkgs-review pr 216511` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.pytest-ansible</li>
    <li>python311Packages.pytest-ansible</li>
  </ul>
</details>
<details>
  <summary>11 packages failed to build:</summary>
  <ul>
    <li>python311Packages.ansible</li>
    <li>python311Packages.ansible-compat</li>
    <li>python311Packages.ansible-core</li>
    <li>python311Packages.ansible-kernel</li>
    <li>python311Packages.ansible-later</li>
    <li>python311Packages.ansible-lint</li>
    <li>python311Packages.ansible-runner</li>
    <li>python311Packages.geoip2</li>
    <li>python311Packages.parsedmarc</li>
    <li>python311Packages.sopel</li>
    <li>python311Packages.ttp</li>
  </ul>
</details>
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>ansible (ansible_2_14 ,python310Packages.ansible-core)</li>
    <li>ansible-later (python310Packages.ansible-later)</li>
    <li>ansible-lint (python310Packages.ansible-lint)</li>
    <li>ansible_2_12</li>
    <li>ansible_2_13</li>
    <li>kargo</li>
    <li>parsedmarc (python310Packages.parsedmarc)</li>
    <li>python310Packages.ansible</li>
    <li>python310Packages.ansible-compat</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.ansible-runner</li>
    <li>python310Packages.geoip2</li>
    <li>python310Packages.mocket</li>
    <li>python310Packages.sopel</li>
    <li>ttp (python310Packages.ttp)</li>
    <li>python311Packages.mocket</li>
  </ul>
</details>

**Comment by hesiod** [comment]
After more fiddling, I think the root issue is simply some Python 3.11 incompatibility in the test setup of mocket (althought I'm not sure what the cause is). Anyways, in the newest version of the commits I'm conditionally disabling certain tests on Python >= 3.11 in mocket and geoip2 - the latter has similar 3.11 issues that weren't uncovered previously because mocket didn't build.

**Comment by davidak** [comment]
Are you aware of https://github.com/NixOS/nixpkgs/pull/216168?

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 216511` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.pytest-ansible</li>
    <li>python311Packages.pytest-ansible</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python311Packages.sopel</li>
  </ul>
</details>
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>ansible (ansible_2_14 ,python310Packages.ansible-core)</li>
    <li>ansible-later (python310Packages.ansible-later)</li>
    <li>ansible-lint (python310Packages.ansible-lint)</li>
    <li>ansible_2_12</li>
    <li>ansible_2_13</li>
    <li>kargo</li>
    <li>parsedmarc (python310Packages.parsedmarc)</li>
    <li>python310Packages.ansible</li>
    <li>python310Packages.ansible-compat</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.ansible-runner</li>
    <li>python310Packages.geoip2</li>
    <li>python310Packages.mocket</li>
    <li>python310Packages.sopel</li>
    <li>ttp (python310Packages.ttp)</li>
    <li>python311Packages.ansible</li>
    <li>python311Packages.ansible-compat</li>
    <li>python311Packages.ansible-core</li>
    <li>python311Packages.ansible-kernel</li>
    <li>python311Packages.ansible-later</li>
    <li>python311Packages.ansible-lint</li>
    <li>python311Packages.ansible-runner</li>
    <li>python311Packages.geoip2</li>
    <li>python311Packages.mocket</li>
    <li>python311Packages.parsedmarc</li>
    <li>python311Packages.ttp</li>
  </ul>
</details>

**Review by davidak — APPROVED** [review]
Changes look good

Build successful

**Comment by fabaff** [comment]
#217295 contains the update of `mocket` to 3.11.0 which solved the build failure of `mocket` itself.

File: pkgs/development/python-modules/geoip2/default.nix [hunk 2]
```diff
@@ -30,6 +30,8 @@ buildPythonPackage rec {
     pytestCheckHook
   ];
 
+  disabledTests = lib.optional (pythonAtLeast "3.11") "TestAsyncClient";
```
Comment (fabaff) [inline pkgs/development/python-modules/geoip2/default.nix:33]: For me there was also a failure on Python 3.10, see https://github.com/NixOS/nixpkgs/pull/217295/commits/e2be01564248989a4dd2192f8410b3e62fd80f9b

**Comment by NickCao** [comment]
Superseded by later PRs.

**Pull request closed without merging**